src/main/kotlin/id/walt/services/vc/WaltIdJwtCredentialService.kt

```diff
@@ -0,0 +1,121 @@
+package id.walt.services.vc
+
+import com.nimbusds.jwt.JWTClaimsSet
+import id.walt.common.prettyPrint
+import id.walt.crypto.KeyAlgorithm
+import id.walt.crypto.SignatureType
+import id.walt.services.essif.EssifServer.nonce
+import id.walt.services.essif.TrustedIssuerClient.domain
+import id.walt.services.jwt.JwtService
+import id.walt.services.keystore.KeyStoreService
+import id.walt.signatory.ProofConfig
+import id.walt.vclib.Helpers.encode
+import id.walt.vclib.Helpers.toCredential
+import id.walt.vclib.Helpers.toMap
+import id.walt.vclib.model.Proof
+import id.walt.vclib.model.VerifiableCredential
+import id.walt.vclib.vclist.Europass
+import info.weboftrust.ldsignatures.LdProof
+import mu.KotlinLogging
+import java.nio.file.Files
+import java.nio.file.Path
+import java.text.SimpleDateFormat
+import java.time.Instant
+import java.util.*
+
+private val log = KotlinLogging.logger {}
+
+open class WaltIdJwtCredentialService : JwtCredentialService() {
+
+    private var ks = KeyStoreService.getService()
+    private val jwtService = JwtService.getService()
+
+    companion object {
+        private val simpleDateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    }
+
+    override fun sign(
+        jsonCred: String,
+        config: ProofConfig
+    ): String {
+        log.debug { "Signing JWT object with: issuerDid (${config.issuerDid}), domain (${config.domain}), nonce (${config.nonce}" }
+        val credential = jsonCred.toCredential()
+
+        val type = ks.load(config.issuerDid).let {
+            when (it.algorithm) {
+                KeyAlgorithm.ECDSA_Secp256k1 -> SignatureType.EcdsaSecp256k1Signature2019.name
+                KeyAlgorithm.EdDSA_Ed25519 -> SignatureType.Ed25519Signature2018.name
+                // else -> throw Exception("Signature for key algorithm ${it.algorithm} not supported")
+            }
+        }
+
+        val jws = when (credential) {
```

Comment:
I am not satified with this part as it violates the Open/Close Principle. @philpotisk I would like to discuss it further if possible